ui/pages/permissions-connect/snaps/snap-install/snap-install.js

```diff
@@ -116,7 +114,7 @@ export default function SnapInstall({
       backgroundColor={BackgroundColor.backgroundAlternative}
     >
       {(isLoading || hasError) && !isOriginSnap ? (
-        <PermissionConnectHeader origin={origin} iconUrl={iconUrl} />
+        <PermissionConnectHeader subjectMetadata={targetSubjectMetadata} />
```

Comment:
Since we only cache the subject metadata inside this component, this fix won't work for Snaps.

I'm assuming this bug can be reproduced using the Ethereum provider example Snap